datalad/interface/add_archive_content.py

```diff
@@ -78,87 +77,107 @@
 class AddArchiveContent(Interface):
     """Add content of an archive under git annex control.
 
-    This results in the files within archive (which must be already under annex
-    control itself) added under annex referencing original archive via
-    custom special remotes mechanism
-
-    Example:
-
-        annex-repo$ datalad add-archive-content my_big_tarball.tar.gz
+    Given an archive under annex control, extract and add its files to the
+    dataset, and reference the original archive as a custom special remote.
 
     """
+    _examples_ = [
+        dict(text="""Add files from the archive 'big_tarball.tar.gz', but
+                     keep big_tarball.tar.gz in the index""",
+             code_py="add_archive_content(path='big_tarball.tar.gz')",
+             code_cmd="datalad add-archive-content big_tarball.tar.gz"),
+        dict(text="""Add files from the archive 'tarball.tar.gz', and
+                     remove big_tarball.tar.gz from the index""",
+             code_py="add_archive_content(path='big_tarball.tar.gz', delete=True)",
+             code_cmd="datalad add-archive-content big_tarball.tar.gz --delete"),
+        dict(text="""Add files from the archive 's3.zip' but remove the leading
+                     directory""",
+             code_py="add_archive_content(path='s3.zip', strip_leading_dirs=True)",
+             code_cmd="datalad add-archive-content s3.zip --strip-leading-dirs"),
+        ]
+
     # XXX prevent common args from being added to the docstring
     _no_eval_results = True
     _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc=""""specify the dataset to save""",
+            constraints=EnsureDataset() | EnsureNone()),
         delete=Parameter(
-            args=("-d", "--delete"),
+            args=("-D", "--delete"),
             action="store_true",
-            doc="""flag to delete original archive from the filesystem/git in current tree.
-                   %s""" % _KEY_OPT_NOTE),
+            doc="""delete original archive from the filesystem/Git in current
+            tree. %s""" % _KEY_OPT_NOTE),
         add_archive_leading_dir=Parameter(
             args=("--add-archive-leading-dir",),
             action="store_true",
-            doc="""flag to place extracted content under a directory which would correspond
-                   to archive name with suffix stripped.  E.g. for archive `example.zip` its
-                   content will be extracted under a directory `example/`"""),
+            doc="""place extracted content under a directory which would
+            correspond to the archive name with suffix stripped. E.g. the
+            content of `example.zip` will be extracted under `example/`"""),
```

Comment:
What about the `.tar.gz` case? Would it be `archivename.tar`? If not, we should maybe say "all suffixes stripped".

Reply:
yes, this is a better wording! I will strip all suffixes.